segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java

```diff
@@ -701,9 +711,48 @@ public void close() {
             } catch (Exception e) {
                 throw new CompletionException(e);
             }
+        }, this.executor)
+        .whenCompleteAsync((v, e) -> {
+            if (shouldRelease.get()) {
```

Comment:
When is this ever not true?

Reply:
when we throw and exception because of Concurrent access . We don't want it to clear up the entry for inflight valid call and unintentionally allow concurrent calls.